### PR TITLE
[Synthetics] Extract Tunnel client API

### PR DESCRIPTION
**Pull request by lefebvree**
### What and why?

Extract the Synthetics `Tunnel` client to its own directory to be used as a standalone module free of external datadog-ci imports.

### How?

Update the `Tunnel` API:
- The optional proxy to use for the tunnel must now be instantiated before and the agent passed to the tunnel constructor.
  `getProxyAgent()` is called before initializing the tunnel in `executeTests()`.
- A `TunnelReporter` is now used which will emit the raw message logs, the default reporter expose a `getTunnelReporter()` utility function to format the logs with chalk.
- Move all tunnel files to the `tunnel/` directory and export the main class and interfaces in its `index.ts`.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/synthetics/tunnel/tunnel.ts [hunk 1]
```diff
@@ -30,24 +28,30 @@ export interface TunnelInfo {
...
-    this.logWarning = (message: string) => reporter.log(`[${chalk.bold.yellow('Tunnel')}] ${message}\n`)
+  private connected = false
+  private ws: WebSocket
+  private multiplexer?: Multiplexer
+  private forwardedSockets: Set<Socket> = new Set()
 
+  constructor(
+    private url: string,
+    private testIDs: string[],
+    proxyAgent?: ReturnType<typeof ProxyAgent>,
```
Comment (jakepruitt) [inline src/commands/synthetics/tunnel/tunnel.ts:52]: Why is `ReturnType<typeof ...>` needed here ?

File: src/commands/synthetics/tunnel/tunnel.ts [hunk 2]
```diff
@@ -30,24 +28,30 @@ export interface TunnelInfo {
...
+    private reporter?: TunnelReporter
```
Comment (jakepruitt) [inline src/commands/synthetics/tunnel/tunnel.ts:53]: Why make the reporter optional ?

**Review by jakepruitt — APPROVED** [review]
Small nits, otherwise looks good to me!

File: src/commands/synthetics/tunnel/tunnel.ts (hunk 1 above)
Comment (lefebvree) [reply src/commands/synthetics/tunnel/tunnel.ts:52]: `ProxyAgent` is actually the constructor and the `proxy-agent` doesn't seem to expose the actual `ProxyAgent` type sadly

File: src/commands/synthetics/tunnel/tunnel.ts (hunk 2 above)
Comment (lefebvree) [reply src/commands/synthetics/tunnel/tunnel.ts:53]: I was hesitant making it optional, some integrations may not need it but maybe we can instead make its methods optional 🤔

Comment (jakepruitt) [reply src/commands/synthetics/tunnel/tunnel.ts:53]: Ah gotcha, this was with integrations in mind. Then yeah, from a core functionality standpoint the reporter is not necessarily "required".

**Pull request merged**